client/servicediscovery/callbacks.go

```diff
@@ -0,0 +1,104 @@
+// Copyright 2025 TiKV Project Authors.
```

Comment:
How about moving this file to pkg?

Reply:
Its code is highly related with the service discovery, `pkg` is for those components which are more common.